addons/l10n_be_invoice_bba/models/account_invoice.py

```diff
@@ -151,55 +147,37 @@ def create(self, cr, uid, vals, context=None):
             if self.check_bbacomm(reference):
                 reference = re.sub('\D', '', reference)
                 vals['reference'] = '+++' + reference[0:3] + '/' + reference[3:7] + '/' + reference[7:] + '+++'
-                same_ids = self.search(cr, uid,
-                    [('type', '=', 'out_invoice'), ('reference_type', '=', 'bba'),
-                     ('reference', '=', vals['reference'])])
+                same_ids = self.search([('type', '=', 'out_invoice'), ('reference_type', '=', 'bba'), ('reference', '=', vals['reference'])])
                 if same_ids:
                     raise UserError(_('The BBA Structured Communication has already been used!' \
                                         '\nPlease create manually a unique BBA Structured Communication.'))
-        return super(account_invoice, self).create(cr, uid, vals, context=context)
+        return super(AccountInvoice, self).create(vals)
 
-    def write(self, cr, uid, ids, vals, context=None):
-        if isinstance(ids, (int, long)):
-            ids = [ids]
-        for inv in self.browse(cr, uid, ids, context):
+    @api.multi
+    def write(self, vals):
+        for invoice in self:
             if vals.has_key('reference_type'):
                 reference_type = vals['reference_type']
             else:
-                reference_type = inv.reference_type or ''
+                reference_type = invoice.reference_type or ''
 
             if reference_type == 'bba' and 'reference' in vals:
                 if self.check_bbacomm(vals['reference']):
                     reference = re.sub('\D', '', vals['reference'])
                     vals['reference'] = '+++' + reference[0:3] + '/' + reference[3:7] + '/' + reference[7:] + '+++'
-                    same_ids = self.search(cr, uid,
-                        [('id', '!=', inv.id), ('type', '=', 'out_invoice'),
+                    same_ids = self.search([('id', '!=', invoice.id), ('type', '=', 'out_invoice'),
                          ('reference_type', '=', 'bba'), ('reference', '=', vals['reference'])])
                     if same_ids:
                         raise UserError(_('The BBA Structured Communication has already been used!' \
                                             '\nPlease create manually a unique BBA Structured Communication.'))
-        return super(account_invoice, self).write(cr, uid, ids, vals, context)
+        return super(AccountInvoice, self).write(vals)
 
-    def copy(self, cr, uid, id, default=None, context=None):
+    @api.multi
+    def copy(self, default=None):

```

Comment:
Yep you're right. It's not mandatory.

On 07/08/2016 11:04 AM, Atul Patel wrote:
>
> In addons/l10n_be_invoice_bba/models/account_invoice.py 
> <https://github.com/odoo/odoo/pull/12724#discussion_r70045196>:
>
> >
> > -    def copy(self, cr, uid, id, default=None, context=None):
> > +    @api.multi
> > +    def copy(self, default=None):
>
> @tivisse <https://github.com/tivisse> ,
> i think copy method always accept single record. don't required 
> explicitly ensure_one..
> see this :
> https://github.com/odoo/odoo/blob/master/addons/mail/models/mail_template.py#L229
>
> https://github.com/odoo/odoo/blob/master/addons/project/models/project.py#L230
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/odoo/odoo/pull/12724/files/ca16b809cf9290ff6accc3198f5f4c07e0f1bba0..9777f8694c879c10a2eca510dcd4a960221288ec#r70045196>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AIC4p-1xmDphGWdt-_XJVUzxnSWOrYbfks5qThKEgaJpZM4JHIm1>.
>

-- 
TIVISSE Yannick
User Experience Coordinator
odoo
+32 81 81 37 00
yti@odoo.com
Chaussée de Namur, 40
1367, Grand-Rosière (BELGIUM)